src/Aspire.Hosting.Azure.KeyVault/api/Aspire.Hosting.Azure.KeyVault.cs

```diff
@@ -11,17 +11,34 @@ namespace Aspire.Hosting
     public static partial class AzureKeyVaultResourceExtensions
     {
         public static ApplicationModel.IResourceBuilder<Azure.AzureKeyVaultResource> AddAzureKeyVault(this IDistributedApplicationBuilder builder, string name) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<T> WithRoleAssignments<T>(this ApplicationModel.IResourceBuilder<T> builder, ApplicationModel.IResourceBuilder<Azure.AzureKeyVaultResource> target, params global::Azure.Provisioning.KeyVault.KeyVaultBuiltInRole[] roles)
+            where T : ApplicationModel.IResource { throw null; }
     }
 }
 
 namespace Aspire.Hosting.Azure
 {
-    public partial class AzureKeyVaultResource : AzureProvisioningResource, ApplicationModel.IResourceWithConnectionString, ApplicationModel.IResource, ApplicationModel.IManifestExpressionProvider, ApplicationModel.IValueProvider, ApplicationModel.IValueWithReferences
+    public partial class AzureKeyVaultResource : AzureProvisioningResource, ApplicationModel.IResourceWithConnectionString, ApplicationModel.IResource, ApplicationModel.IManifestExpressionProvider, ApplicationModel.IValueProvider, ApplicationModel.IValueWithReferences, IKeyVaultResource, ApplicationModel.IAzureResource
     {
         public AzureKeyVaultResource(string name, System.Action<AzureResourceInfrastructure> configureInfrastructure) : base(default!, default!) { }
 
+        BicepOutputReference IKeyVaultResource.IdOutputReference { get { throw null; } }
+
+        System.Func<string, System.Threading.CancellationToken, System.Threading.Tasks.Task<string?>>? IKeyVaultResource.SecretResolver { get { throw null; } set { } }
+
+        BicepOutputReference IKeyVaultResource.VaultUriOutputReference { get { throw null; } }
+
         public ApplicationModel.ReferenceExpression ConnectionStringExpression { get { throw null; } }
 
+        public BicepOutputReference Id { get { throw null; } }
```

Comment:
@davidfowl - why do we need the `Id`? Shouldn't the `VaultUri` be enough?

Looking at the references in code. I'm not seeing where this is used.

Reply:
Opened [Remove IKeyVaultResource.IdOutputReference from public API (dotnet/aspire#8352)](https://github.com/dotnet/aspire/pull/8352)